src/p2G4_func_queue.c

```diff
@@ -77,26 +100,60 @@ void fq_add(bs_time_t time, f_index_t index, uint32_t dev_nbr) {
 void fq_remove(uint32_t d){
   f_queue[d].f_index = None;
   f_queue[d].time = TIME_NEVER;
+  f_queue[d].pend = false;
+}
+
+/**
+ * Advance the function queue to the next element for the current time value
+ * If time advances, re-sort the function queue
+ */
+void fq_step(bs_time_t current_time) {
+  if ( !force_sort ) {
+    next_idx++;
+    if (current_time == f_queue[device_list[next_idx]].time) {
+      return;
+    }
+  }
+  force_sort = false;
+  fq_sort();
 
-  fq_find_next();
+  if ( flush_pend ) {
+    flush_pend = false;
+
+    uint32_t head_dev = device_list[0];
+
+    bs_time_t now = f_queue[head_dev].time;
+    for (int i = 0; i < n_devs; i++) {
```

Comment:
```suggestion
    for (uint32_t i = 0; i < n_devs; i++) {
```
Similarly for other uses for `(i < n_devs)`